networkit/_NetworKit.pyx

```diff
@@ -6158,11 +6171,25 @@ cdef class ConnectedComponents(Algorithm):
 		component_sizes = cc_result.component_sizes
 		n_components = cc_result.n_components
 		array_wrapper = ArrayWrapper()
-		np_mapping_array = array_wrapper.as_ndarray(n_nodes, <void *>components)
+		np_mapping_array = array_wrapper.as_ndarray(n_nodes, <void *>components, np.NPY_UINT64)
+		array_wrapper = ArrayWrapper()
+		np_component_sizes = array_wrapper.as_ndarray(n_components, <void *>component_sizes, np.NPY_UINT64)
+
+		# wrap the equivalence classes into a numpy array of numpy arrays
+		cdef void ** equivalence_classes = <void**> cc_result.equivalence_classes
+		for i in range(n_components):
+			array_wrapper = ArrayWrapper()
+			size = component_sizes[i]
+			row = equivalence_classes[i]
+			ndarray = array_wrapper.as_ndarray(size, row, np.NPY_UINT64)
+			# increase ref count for python objects to prevent them from deletion
+			Py_INCREF(ndarray)
```

Comment:
We should rather move the `INCREF` into `as_ndarray` of `ArrayWrapper`. The wrapper object should the have an internal counter for the references, and decrease by them in the destructor.

Reply:
As discussed in person, `ndarray` is already garbage collected before `np_equivalence_classes` is constructed. So increasing the reference count must happen here already :+1: